src/odemis/driver/autoscript_client.py

```diff
@@ -0,0 +1,2065 @@
+            self.server._pyroClaimOwnership()
+            self.server.set_default_application_file(application_file)
+
+    def set_default_patterning_beam_type(self, channel: str) -> None:
+        """
+        Set the default patterning beam type.
+        :param channel: Name of one of the channels.
+        """
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            self.server.set_default_patterning_beam_type(channel)
+
+    def get_available_application_files(self) -> List[str]:
+        """Returns: the available application files."""
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            return self.server.get_available_application_files()
+
+    def estimate_milling_time(self) -> float:
+        """Returns: the estimated milling time for currently drawn patterns."""
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            return self.server.estimate_milling_time()
+
+
+class Scanner(model.Emitter):
+    """
+    This is an extension of the model.Emitter class. It contains Vigilant
+    Attributes for magnification, accel voltage, blanking, spotsize, beam shift,
+    rotation and dwell time. Whenever one of these attributes is changed, its
+    setter also updates another value if needed.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        role: str,
+        parent: SEM,
+        hfw_nomag: float,
+        channel: str,
+        has_detector: bool = False,
+        **kwargs,
+    ):
+        """
+        :param name: name of the Scanner
+        :param role: role of the Scanner
+        :param parent: parent of the Scanner
+        :param hfw_nomag: horizontal field width at nominal magnification
+        :param channel: name of the electron channel
+        :param has_detector: True if a Detector is also controlled. In this case,
+          the .resolution, .scale and associated VAs will be provided too.
+        """
+        self.parent: SEM  # for type hinting
+        model.Emitter.__init__(self, name, role, parent=parent, **kwargs)
+
+        self.channel = channel  # name of the channel used
+
+        # will take care of executing auto contrast/brightness and auto stigmator asynchronously
+        self._executor = CancellableThreadPoolExecutor(max_workers=1)  # one task at a time
+
+        self._hfw_nomag = hfw_nomag
+        self._has_detector = has_detector
+        self._acq_type = model.MD_AT_EM if channel == "electron" else model.MD_AT_FIB
+
+        # dwell time
+        dwell_time_info = self.parent.dwell_time_info(self.channel)
+        self.dwellTime = model.FloatContinuous(
+            self.parent.get_dwell_time(self.channel),
+            dwell_time_info["range"],
+            unit=dwell_time_info["unit"],
+            setter=self._setDwellTime)
+        # when the range has changed, clip the current dwell time value to the new range
+        self.dwellTime.clip_on_range = True
```

Comment:
Please move dwellTime to the "has_detector" part. It doesn't make sense to control the dwell time if there is no acquisition.

Reply:
I don't agree with this, see other client PR